src/app/pages/client/questions/questions.page.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnInit } from '@angular/core';
+import { QuestionService } from 'src/app/services/question.service';
+import {ActivatedRoute, Router} from '@angular/router';
+import { NotificationService } from 'src/app/services/notification.service';
+
+@Component({
+  selector: 'app-questions',
+  templateUrl: './questions.page.html',
+  styleUrls: ['./questions.page.scss'],
+})
+export class QuestionsPage implements OnInit {
+
+  availableQuestions: any[] = [];
+  answeredQuestions : any[] = [];
+
+  constructor(
+    private questions   : QuestionService, 
+    private route       : ActivatedRoute,
+    private router      : Router,
+    private notification: NotificationService
+  ) {
+    this.route.queryParams.subscribe(params => {
+      if (this.router.getCurrentNavigation().extras.state) {
+        let questionId = this.router.getCurrentNavigation().extras.state.questionId;
+        console.log(`Question ${questionId} was answered`);
+        this.updateQuestions();
+      }
+    });
+  }
+
+  /**
+   * Updates the list of question
+   */
+  private async updateQuestions(){
```

Comment:
On error with `this.questions.getQuestions()` returning an empty array. `this.availableQuestions` and `this.answeredQuestions` will be cleaned without receiving new questions. It would be good to add this event to function doc as an implementation note or to change this behavior.

Reply:
I think the result of `this.questions.getQuestions()` should be trusted as it comes from the server. If the server replies without sending any questions, we should assume there are no questions for a valid reason.

I updated the doc in c275629